### PR TITLE
Add guiding principles

### PR DESCRIPTION
**Pull request by BradSwain**

### CODE REVIEW

**Review by funemy — COMMENTED** [review]
LOL, overall I like it. But the metrics are vague, hard to enforce it?

**Comment by BradSwain** [comment]
yeah. Not really something to enforce. Just some things to aim for.

For precision we can manually check reports on some projects.
For coverage we can (and plan to) add some checks of how many functions are scanned vs. how many are in the IR (or something along those lines.)
For scalability we can use CPU and heap profiling tools to roughly track  performance as we make updates.

**Comment by funemy** [comment]
> yeah. Not really something to enforce. Just some things to aim for.
> 
> For precision we can manually check reports on some projects.
> For coverage we can (and plan to) add some checks of how many functions are scanned vs. how many are in the IR (or something along those lines.)
> For scalability we can use CPU and heap profiling tools to roughly track performance as we make updates.

Sounds good to me. I think the coverage and profiling both sound intriguing.

File: README.md [hunk 1]
```diff
@@ -15,6 +15,41 @@ OpenRace is an open source implementation of the [Coderrect Scanner][CS] which i
...
+The main benefit of static analysis over dynamic analysis is higher code coverage.
+
+We should ensure that our tool is scanning as much of a given code base as is possible.
+
+### Scalability
+Eventually we plan for the tool to be integrated into CI/CD. For most people, this means that the tool must run on fairly limited resources.
+
+"Overnight" scale means if the tool starts running at the end of the day, the results are ready the next morning.
+
+"Desktop" scale means the tool should not use more resources than are available on commodoty desktops.
```
Comment (funemy) [inline README.md:46]: commodity?

File: README.md [hunk 2]
```diff
@@ -15,6 +15,41 @@ OpenRace is an open source implementation of the [Coderrect Scanner][CS] which i
...
 
+## Guiding Principles
+
+When making design or implementation decisions, we should try to prioritize the following three goals.
+
+1. **Precision**: 2 out of 3 reported races should be true positives. 
+2. **Coverage**: On average, the tool should scan the majority of a code base. 
+3. **Scalability**: The tool should run at "overnight" and "desktop" scale, even for large projects. 
+
+These are goals we should do our best to achieve, but are not necesarily hard requirements.
```
Comment (funemy) [inline README.md:26]: necessarily

**Review by funemy — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (BradSwain) [reply README.md]: Meant to type common. Updated.

File: README.md (hunk 2 above)
Comment (BradSwain) [reply README.md]: fixed

**Review by funemy — APPROVED** [review]

**Pull request merged**